docs/guides/universal-receiver-delegate/set-default-implementation.md

```diff
@@ -11,61 +14,289 @@ Users deploying their Universal Profiles using the guides that utilize **[lsp-fa
+    });
+
+  return universalProfileURDAddress;
+};
+
+// deploy a new Universal Profile URD and retrieve its address
+const universalProfileURDAddress = await deployUniversalProfileURD();
+```
+
+  </TabItem>
+
+  <TabItem value="ethersjs" label="ethers.js">
+
+```typescript title="Deploy a Universal Receiver Delegate for the Universal Profile"
+const deployUniversalProfileURD = async () => {
+  // create a LSP1UniversalReceiverDelegateUP Contract Factory
+  let universalProfileURDFactory = new ethers.ContractFactory(
+    LSP1UniversalReceiverDelegateUP.abi,
+    LSP1UniversalReceiverDelegateUP.bytecode,
+  );
+
+  // deploy the Universal Receiver Delegate UP contract
+  const universalProfileURD = await universalProfileURDFactory
+    .connect(EOA)
+    .deploy();
+
+  return universalProfileURD.target;
+};
+
+// deploy a new Universal Profile URD and retrieve its address
+const universalProfileURDAddress = await deployUniversalProfileURD();
+```
+
+  </TabItem>
+
+</Tabs>
+
+## Step 3 - Set the address of the URD in the storage
+
+After deploying the contract, we need to set its address under the **[LSP1-UniversalReceiverDelegate Data Key](../../standards/generic-standards/lsp1-universal-receiver.md#extension)** and grant it the **[SUPER_SETDATA](../../standards/universal-profile/lsp6-key-manager.md#super-permissions)** permission.
+
+### Step 3.1 - Create the contract instances
+
+Firstly we need to create instances for the following contracts:
+
+- [**Universal Profile**](../../standards/universal-profile/lsp0-erc725account.md)
+- [**Key Manager**](../../standards/universal-profile/lsp6-key-manager.md)
+
+<Tabs>
+  
+  <TabItem value="web3js" label="web3.js">
+
+```typescript title="Contract instances for the Universal Profile & Key Manager"
+// create an insatnce of the Universal Profile
+const universalProfile = new web3.eth.Contract(
   UniversalProfile.abi,
-  myUniversalProfileAddress,
+  universalProfileAddress,
 );
+// get the owner of the Universal Profile
+// in our case it should be the address of the Key Manager
+const keyManagerAddress = await universalProfile.methods.owner().call();
+// create an instance of the LSP6KeyManager
+const keyManager = new web3.eth.Contract(LSP6KeyManager.abi, keyManagerAddress);
+```
+
+  </TabItem>
+
+  <TabItem value="ethersjs" label="ethers.js">
+
+```typescript title="Contract instances for the Universal Profile & Key Manager"
+// create an insatnce of the Universal Profile
+const universalProfile = new ethers.Contract(
+  universalProfileAddress,
+  UniversalProfile.abi,
+);
+// get the owner of the Universal Profile
+// in our case it should be the address of the Key Manager
+const keyManagerAddress = await universalProfile.methods.owner().call();
+// create an instance of the LSP6KeyManager
+const keyManager = new ethers.Contract(keyManagerAddress, LSP6KeyManager.abi);
+```
 
+  </TabItem>
+
+</Tabs>
+
+### Step 3.2 - Encode new permissions Data Keys & Values
```

Comment:
this sub-heading only mention the permissions.

Maybe we can also mention the fact that we are registering it on the UP in the title as well?

Something like:

```suggestion
### Step 3.2 - Register URD on the UP + set the URD permissions
```